docs/src/guide/invariants/index.md

```diff
@@ -0,0 +1,330 @@
+# {#guide-invariants} Writing Invariants
+
+Invariants are an essential feature when writing complex Reach DApps. 
+Our guide @{seclink("guide-loop-invs")} provides a deep theoretical understanding of loop invariants, and the @{seclink("tut")} and @{seclink("tut-rsvp")} tutorials both feature invariants. If you desire to create a complex Reach DApp, then you'll likely need to write your own invariants, as well.
+
+To help prepare you to write your own invariants, we'll review a few examples and break them to help understand the error messages. 
+At the end of the guide you'll have a respect for how important invariants are in Reach Applications and how important it is to have a complete understanding of your application before writing the codebase. 
+
+First, a few basics about invariants. 
+
+Invariant refers to something that is constant without variation. 
+
+Loop invariants are required when writing Reach `{!rsh} while` loops.
+They are established immediately before the loop begins. 
+You are allowed to write as many `{!rsh} invariant` statements as needed. 
+Inside a `{!rsh} parallelReduce`, use dot notation to place each `{!rsh} invariant` on its own line to easily distinguish each `{!rsh} invariant` from the other.
```

Comment:
A) I think this is a weird time to say this

B) This would be more clear with a comparison of actual code